injection/sharedmain/main.go

```diff
@@ -59,10 +59,25 @@ func init() {
 	maxprocs.Set()
 }
 
-// GetLoggingConfig gets the logging config from either the file system if present
-// or via reading a configMap from the API.
+type lcfg struct{}
+
+// WithLoggingConfig associates a logging configuration with the context.  This
+// configuration takes priority in GetLoggingConfig (below).
+func WithLoggingConfig(ctx context.Context, cfg *logging.Config) context.Context {
```

Comment:
Might make sense to move these helpers (With*, Get*) to `pkg/logging`.

Though the API lookup should probably remain in `sharedmain`

Reply:
Done for both